packages/app-desktop/gui/MainScreen/commands/notesSortOrderSwitch.ts

```diff
@@ -0,0 +1,88 @@
+import { CommandContext, CommandDeclaration, CommandRuntime } from '@joplin/lib/services/CommandService';
+import Setting from '@joplin/lib/models/Setting';
+import { _ } from '@joplin/lib/locale';
+
+let fields: string[] = null;
+let perFieldReverse: { [field: string]: boolean } = null;
+
+export const notesSortOrderFieldArray = (): string[] => {
+	// The order of the fields is strictly determinate.
+	if (fields == null) {
+		fields = Setting.enumOptionValues('notes.sortOrder.field').sort().reverse();
+	}
+	return fields;
+};
+
+export const notesSortOrderNextField = (currentField: string) => {
+	const fields = notesSortOrderFieldArray();
+	const index = fields.indexOf(currentField);
+	if (index < 0) {
+		return currentField;
+	} else {
+		return fields[(index + 1) % fields.length];
+	}
+};
+
+export const declaration: CommandDeclaration = {
+	name: 'notesSortOrderSwitch',
```

Comment:
By the way commands are normally named as `verbNoun` and we use the word "toggle" usually. So it should be name "toggleNoteSortOrder" for consistency.

Reply:
I see. I have no objection to `verbNoun` style and other naming conventions. Speaking my preferences, "toggle" would be limited to switching between two states, since its behavior is very self-explanatory. For three or more states, I prefer other verbs, switch, select, change, or cycle.

Reply:
Finished. 36e00f8
Command names are changed as below.
- notesSortOrderSwitch ->  toggleNotesSortOrderField
- notesSortOrderToggleReverse -> toggleNotesSortOrderReverse
- perFolderSortOrderSwitch -> togglePerFolderSortOrder